File: smp2017-android-muskan/cs50_psets/pset2/Muskan2108-cs50-2017-x-caesar/caesar.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>

int main( int argc, string argv[] )  {

   if( argc == 2 ) 
{	string s;
	printf("plaintext: ");
	s=get_string();
	int n;
	n=strlen(s);
	long long int u,t;
	u=atoi(argv[1]);
	u=u%26;
	printf("ciphertext: ");
	for(int i=0;i<n;i++)
	{
		if((s[i]>=65)&&(s[i]<=90))
		{
			t=s[i]+u;
			if(t>90)
			{	t=64+(t%90);	}
			s[i]=t;
			printf("%c",s[i]);	
		}
		else if((s[i]>=97)&&(s[i]<=122))
		{
			t=s[i]+u;
			if(t>122)
			{	t=96+(t%122);	}
			s[i]=t;
			printf("%c",s[i]);	
		}
		else
		{	printf("%c",s[i]);	}
	}
	printf("\n");
return 0;
	
   }
   else if( argc > 2 ) {
      printf("Too many arguments supplied.\n");
	return 1;
   }
   else {
      printf("One argument expected.\n");
	return 1;
 	}		
}

File: smp2017-android-muskan/cs50_psets/pset5/pset6/Muskan2108-cs50-2017-x-pset6/caesar.py
import cs50
import sys
def main():
    if len(sys.argv)>2:
        print("Too many arguments supplied")
        exit(1)
    elif len(sys.argv)<=1:
       print("One argument expected")
       exit(1)
    else:
       k=int(sys.argv[1])
       k=k%26
       cipher=[]
       print("plaintext:",end="")
       plain=cs50.get_string()
       for i in plain:
           if i.isalpha():
               cipher.append(caesar(i,k))
           else:
                cipher.append(i)
                
       print("ciphertext:",end="")
       print("".join(cipher))
       exit(0)
def caesar(char,key):
      if char.isupper():
        return chr(((ord(char) - 65 + key) % 26) + 65)
      else:
        return chr(((ord(char) - 97 + key) % 26) + 97)
if __name__ == "__main__":
    main()
    
    